### PR TITLE
Move common functions

### PR DESCRIPTION
**Pull request by AlfHou**
Move functions that are common for optimisation problems to the relevant classes so they can more easily be reused. Also, fix some formatting and typos and move code into a main function.

### CODE REVIEW

File: optimization_task1.py [hunk 1]
```diff
@@ -4,102 +4,76 @@
...
     """
-    left_inequalities = []
-    for i in range(count_appliance):
-        for j in range(hours):
-            lst = [0] * (count_appliance * h)
-            lst[j + hours * i] = 1
-            left_inequalities.append(lst)
-    return left_inequalities
 
+    ineq_matrix = np.identity(HOURS_IN_DAY * len(house_1.get_appliances()))
```
Comment (AlfHou) [inline optimization_task1.py:48]: Just wanted to add that the `max_power_per_hour` generates an identity matrix that is `HOURS_IN_DAY * number of appliances` long, using the numpy identity function is just a bit easier :)

__Old matrix:__
image
__New matrix:__
image

Comment (Elmir-Zeynalov) [reply optimization_task1.py:48]: Yeh, haven't touched python in years so am a little rusty on numpy but the identity function is exactly what we need here :)

**Comment by Elmir-Zeynalov** [comment]
If this is ready then I can approve the pull req and merge into main

**Comment by AlfHou** [comment]
I'll take a last look through it, then I'll take it out of draft 😃

**Review by Elmir-Zeynalov — APPROVED** [review]

**Pull request merged**